ingester/user_state.go

```diff
@@ -165,13 +171,18 @@ func (u *userState) unlockedGet(metric model.Metric) (model.Fingerprint, *memory
 
 	series, ok := u.fpToSeries.get(fp)
 	if ok {
-		return fp, series
+		return fp, series, nil
+	}
+
+	if u.fpToSeries.length() >= maxSeries {
```

Comment:
There's theoretically a (relatively harmless) race here if multiple goroutines get the length of the series map at the same time, then all proceed to add a new series. I'm not sure if it's worth addressing, as this should rarely happen (all samples from one push are added serially), and the possible overshoot in allowed series would probably be minimal.